src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiResponseGenerator.cs

```diff
@@ -191,18 +191,58 @@ public static OpenApiResponse CreateOperationResponse(this ODataContext context,
             OpenApiSchema schema;
             if (operation.ReturnType.IsCollection())
             {
-                schema = new OpenApiSchema
+                OpenApiSchema baseSchema = new()
                 {
-                    Title = operation.ReturnType.Definition.AsElementType() is not IEdmEntityType entityType 
-                        ? null : $"Collection of {entityType.Name}",
-                    Type = "object",
+                    Type = Constants.ObjectType,
                     Properties = new Dictionary<string, OpenApiSchema>
                     {
                         {
                             "value", context.CreateEdmTypeSchema(operation.ReturnType)
                         }
                     }
                 };
+
+                if (operation.IsDeltaFunction())
+                {
+                    if (context.Settings.EnableODataAnnotationReferencesForResponses)
+                    {                       
+                        schema = new OpenApiSchema
+                        {
+                            AllOf = new List<OpenApiSchema>
+                            {
+                                new OpenApiSchema
+                                {
+                                    UnresolvedReference = true,
+                                    Reference = new OpenApiReference
+                                    {
+                                        Type = ReferenceType.Schema,
+                                        Id = Constants.BaseDeltaFunctionResponse  // @odata.nextLink + @odata.deltaLink
+                                    }
+                                },
+                                baseSchema
+                            }
+                        };
+                    }
+                    else
+                    {
+                        baseSchema.Properties.Add(ODataConstants.OdataNextLink);
+                        baseSchema.Properties.Add(ODataConstants.OdataDeltaLink);
+                        schema = baseSchema;
+                    }
+                }
+                else if (context.Settings.EnablePagination)
```

Comment:
Include support for `EnableCount=true` as well?

Reply:
also calling out your attention on this one @millicentachieng in case you missed it.